core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/function/RunningStats.java

```diff
@@ -55,7 +56,10 @@ public double mean() {
    * @return Variance of all numbers.
    */
   public double variance() {
-    return mean2 / (entries - 1.0d);
+    if (numEntries == 1) {
+      return 0;
+    }
+    return mean2 / (numEntries - 1);
```

Comment:
Using (numEntries - 1) means this is calculating a sample variance and not the true variance of the data being read in the pipeline. I don't think this is what we want anyway, and that this should have just been numEntries instead of (numEntires - 1) to begin with.

Reply:
I think that the [source blog](https://www.johndcook.com/blog/skewness_kurtosis/) was using [Bessel's correction](https://en.wikipedia.org/wiki/Bessel%27s_correction) for computing the sample statistic.

I think that even if we use numEntries for the denominator, it won't give us the true variance. Instead, we'll just be accounting for the bias less in the sample statistic.

Reply:
No, the formula for stddev of a sample and of a population just differs by that denominator, whether you subtract 1. Without subtracting 1, you get the actual variance of the data that was processed. Subtracting 1 means you think the data is a sample of a larger population and you are estimating the variance of that population. 

For the purposes of a pipeline, I think it makes more sense to treat it like a population, not as a sample.

Reply:
Ok, I pushed this diff, as well as a change to the test case accordingly:
![image](https://user-images.githubusercontent.com/2440977/52301331-1300aa80-293f-11e9-82ed-c17a664ca028.png)